### DISCUSSION TITLE
How to import audio stimuli from .csv file?

### DISCUSSION THREAD

**Discussion by Cbenaa** [jspsych/jsPsych#693] [Q&A] [answered]
Hello!
So I have two experiments, one with 80 main trials (audio identification task) and one with 152 main trials (audio rating task). Now I know that I can put them in using the regular method but that would take a super long time, is there anyway I could import them through a .csv file and if yes, how would one go about that?

Thanks!

**Comment by vijaymarupudi** [comment]
You can use a tool like https://csvjson.com/csv2json to convert your CSV file to a JSON representation. You can then copy the JSON string to your Javascript file.

**Reply by amunn** [reply]
Another tool that seems quite useful for our workflow (in which students use Google sheets for creating stimulus files) is the following addon for Google sheets. https://github.com/Synthoid/ExportSheetData. To create the right format you need to make sure the following options are clicked: `Export sheet arrays` and  `Export contents as array`.

**Comment by Cbenaa** [comment]
Is there a particular code to use when copy-pasting it or?

**Comment by vijaymarupudi** [comment, marked as answer]
JSON is essentially Javascript code. In the code below, you can paste the JSON where `!!!` is and you can now access your data as a list of objects, one for each row. Each key in those objects is a column name and value is the value for the row.

```javascript
var stimuli = !!!
```

**Comment by becky-gilbert** [comment]
Just to expand on @vijaymarupudi's answer, if your CSV text looks like this:
```
"audio_file","condition"
"1.mp3","1"
"2.mp3","2"
```
Then it will look like this when you use a CSV-to-JSON converter:
```json
[
  {
    "audio_file": "1.mp3",
    "condition": 1
  },
  {
    "audio_file": "2.mp3",
    "condition": 2
  }
]
```
Then, as @vijaymarupudi says, you just need to save this as a JavaScript variable in your jsPsych code:
```javascript
var stimuli = [
  {
    "audio_file": "1.mp3",
    "condition": 1
  },
  {
    "audio_file": "2.mp3",
    "condition": 2
  }
];
```
The above code should work, but this isn't "proper" JavaScript because the object keys are in quotes. So you could fix that by removing the quotes around `audio_file` and `condition` by doing a find-and-replace in your text editor. 

Or, you could keep your CSV-to-JSON result as a JSON-formatted text string by putting quotes around the whole thing, save it as a JavaScript variable, then use `JSON.parse()` to turn it into JavaScript:
```javascript
var json_stimuli = `[
  {
    "audio_file": "1.mp3",
    "condition": 1
  },
  {
    "audio_file": "2.mp3",
    "condition": 2
  }
]`;
var stimuli = JSON.parse(json_stimuli);
```

**Reply by vijaymarupudi** [reply]
@becky-gilbert I don't believe JavaScript allows line breaks in `"` strings, I'll edit to response to use backticks `` ` ``

**Reply by becky-gilbert** [reply]
Oops - thanks @vijaymarupudi !
